src/auto.ts

```diff
@@ -308,14 +312,18 @@ export default class Auto {
     this.logger.verbose.info('Posting comment to GitHub\n', msg);
 
     if (!dryRun) {
-      await this.git.createStatus({
-        ...options,
-        ...msg,
-        target_url,
-        sha
-      } as IPRInfo);
-
-      this.logger.log.success('Posted status to Pull Request.');
+      try {
+        await this.git.createStatus({
+          ...options,
+          ...msg,
+          target_url,
+          sha
+        } as IPRInfo);
+
+        this.logger.log.success('Posted status to Pull Request.');
+      } catch (error) {
+        this.logger.log.error('Failed to post status to github');
```

Comment:
Same thing here. By catching the error and just logging a problem rather than letting the exception bubble up, the cli's exit code will 0, and we won't mark the cmd as failed: https://github.com/intuit/auto/blob/master/src/bin/auto.ts#L10 

Seeing as how these 2 methods' primary function is to set a status on a PR, I feel like we should fail the process if we can't do that.

Reply:
That's a valid point. So should we just throw an exception from here? We specifically _don't_ want to allow the caught error to continue to propagate because it could potentially expose sensitive information still.

Reply:
We could steingify the response, replace the token, parse the string and throw that.

Reply:
Well, the hook should (hopefully) do that. It could be beefed up a little. But is exposing the octokit error to the user really something that should happen anyway? It's not necessarily the friendliest of feedback. We can grab things like the response status code or error message and bubble that up in a new exception.

Reply:
👍 Just throw an error w/ the same message as the log, and attach the response code

Reply:
@zephraph this sound okay?